src/utils/data_buffer.cpp

```diff
@@ -20,15 +38,25 @@ bool DataBuffer::is_ready ()
 
 void DataBuffer::add_data (double *value)
 {
+    if (!is_ready ())
+        return;
+
     lock.lock ();
-    memcpy (this->data + first_free * num_samples, value, sizeof (double) * num_samples);
-    first_free = next (first_free);
-    count++;
-    if (first_free == first_used)
+
+    if (count == 0)
```

Comment:
I am not sure in this and dont remember all this math with indexes now.. size - 1 is not a bug, its by design and there are articles explaining it, lets dont touch it. As far as I understand with counter in addition to indexes it can be avoided but its designed for huge arrays for thousands of elements and there its not important. Also, its internal helper class